src/processor/operator/aggregate/aggregate_hash_table.cpp

```diff
@@ -557,16 +629,24 @@ void AggregateHashTable::updateAggState(const std::vector<ValueVector*>& flatKey
 
 void AggregateHashTable::updateAggStates(const std::vector<ValueVector*>& flatKeyVectors,
     const std::vector<ValueVector*>& unFlatKeyVectors,
-    const std::vector<AggregateInput>& aggregateInputs, uint64_t resultSetMultiplicity) {
+    const std::vector<AggregateInput>& aggregateInputs, uint64_t resultSetMultiplicity,
+    bool updateDistinct) {
     auto aggregateStateOffset = aggStateColOffsetInFT;
     for (auto i = 0u; i < aggregateFunctions.size(); i++) {
-        auto multiplicity = resultSetMultiplicity;
-        for (auto& dataChunk : aggregateInputs[i].multiplicityChunks) {
-            multiplicity *= dataChunk->state->getSelVector().getSelSize();
+        if (!aggregateFunctions[i].isDistinct || updateDistinct) {
+            auto multiplicity = resultSetMultiplicity;
+            for (auto& dataChunk : aggregateInputs[i].multiplicityChunks) {
+                multiplicity *= dataChunk->state->getSelVector().getSelSize();
+            }
+            updateAggFuncs[i](this, flatKeyVectors, unFlatKeyVectors, aggregateFunctions[i],
+                aggregateInputs[i].aggregateVector, multiplicity, i, aggregateStateOffset);
+            aggregateStateOffset += aggregateFunctions[i].getAggregateStateSize();
+        } else {
+            // If a function is distinct we still need to insert the value into the distinct hash
+            // table
+            distinctHashTables[i]->isAggregateValueDistinctForGroupByKeys(flatKeyVectors,
```

Comment:
I don't follow here, don't we need to check the return value here and update the hash table if the key is distinct? Is that done already in `isAggregateValueDistinctForGroupByKeys`? If so, the function name is not accurate.

Reply:
Yes, the `isAggregateValueDistinctForGroupByKeys` function also inserts. I can update the function name to make it more clear.